File: packages/dotnet/Ygo74.AgentRuntime/Domains/Mapping/ResponseMapper.cs
using Ygo74.AgentRuntime.Domains.Contracts;

namespace Ygo74.AgentRuntime.Domains.Mapping;

public static class ResponseMapper
{
    public static IDictionary<string, object?> ToEndpoint(StandardExchangeResponse response)
    {
        return new Dictionary<string, object?>
        {
            ["request_id"] = response.RequestId,
            ["status"] = response.Status,
            ["output"] = response.Output,
            ["error"] = response.Error
        };
    }
}

File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/mapping/ResponseMapper.java
package com.ygo74.agentruntime.domains.mapping;

import com.ygo74.agentruntime.domains.contracts.ExchangeModels.StandardExchangeResponse;

import java.util.HashMap;
import java.util.Map;

public final class ResponseMapper {
    private ResponseMapper() {}

    public static Map<String, Object> toEndpoint(StandardExchangeResponse response) {
        Map<String, Object> mapped = new HashMap<>();
        mapped.put("request_id", response.requestId());
        mapped.put("status", response.status());
        mapped.put("output", response.output());
        mapped.put("error", response.error());
        return mapped;
    }
}
